lib/molinillo/resolution.rb

```diff
@@ -318,62 +322,72 @@ def raise_error_unless_state(conflicts)
 
       # @return [UnwindDetails] Details of the nearest index to which we could unwind
       def build_details_for_unwind
-        # Process the current conflict first, as it's like to produce the highest
-        # index, allowing us to short-circuit subsequent checks
+        # Get the possible unwinds for the current conflict
         current_conflict = conflicts[name]
-        unwind_details = unwind_details_for_conflict(current_conflict)
-        return unwind_details if unwind_details.state_index == states.size - 2
+        binding_requirements = binding_requirements_for_conflict(current_conflict)
+        unwind_details = unwind_options_for_requirements(binding_requirements)
 
-        # Process previous conflicts
-        conflicts.values.each do |conflict|
-          next if conflict == current_conflict
-          unwind_details = unwind_details_for_conflict(conflict, unwind_details)
+        last_detail_for_current_unwind = unwind_details.last
+        current_detail = last_detail_for_current_unwind
+
+        # Look for a past conflict that could be unwound further / differently,
+        # and where doing so would affect the requirement tree for the current
+        # conflict
+        unused_unwind_options.each do |alternative|
+          next unless alternative > current_detail
+          intersecting_requirements =
+            alternative.requirement_trees.flatten(1) &
+            last_detail_for_current_unwind.requirement_trees.flatten(1)
```

Comment:
this can be pulled out into a local, but it might be worth memoizing this in the details?